lib/puppet/type/elbv2_targetgroup.rb

```diff
@@ -0,0 +1,164 @@
+require_relative '../../puppet_x/puppetlabs/property/tag.rb'
+require 'puppet/property/boolean'
+
+Puppet::Type.newtype(:elbv2_targetgroup) do
+  @doc = 'Type representing an ELBv2 target group.'
+
+  ensurable
+
+  newparam(:name, namevar: true) do
+    desc 'The name of the target group.'
+    validate do |value|
+      fail 'Target Groups must have a name' if value == ''
+      fail 'name should be a String' unless value.is_a?(String)
+    end
+  end
+
+  newproperty(:region) do
+    desc 'The region in which to launch the load balancer.'
+    validate do |value|
+      fail 'region must be specified' unless value
+      fail 'region must not contain spaces' if value =~ /\s/
+      fail 'region should be a String' unless value.is_a?(String)
+    end
+  end
+
+  newproperty(:protocol) do
+    desc 'Protocol to use for routing traffic to targets (HTTP/HTTPS)'
+    newvalues(:HTTP, :HTTPS)
+    validate do |value|
+      file 'protocol must be specified' unless value
```

Comment:
Yes, you are correct.
Unfortuately this is NOT something that I'm actively working on at present.